administrator/components/com_installer/Model/UpdatesitesModel.php

```diff
@@ -547,7 +575,43 @@ protected function getListQuery()
 				->bind(':siteId', substr($search, 3), ParameterType::INTEGER);
 		}
 
-		// Note: The search for name, ordering and pagination are processed by the parent InstallerModel class (in extension.php).
+		if ($supported != 0)
+		{
+			switch ($supported)
+			{
+				// Show Update Sites which support Download Keys
+				case 1:
+					$supportedIDs = InstallerHelper::getDownloadKeySupportedSites($enabled);
+					break;
+
+				// Show Update Sites which are missing Download Keys
+				case -1:
+					$supportedIDs = InstallerHelper::getDownloadKeyExistsSites(false, $enabled);
+					break;
+
+				// Show Update Sites which have valid Download Keys
+				case 2:
+					$supportedIDs = InstallerHelper::getDownloadKeyExistsSites(true, $enabled);
+					break;
+			}
+
+			if (!empty($supportedIDs))
+			{
+				$query->whereIn($db->qn('s.update_site_id'), $supportedIDs);
+			}
+			else
+			{
+				// In case of an empty list of IDs we apply a fake filter to effectively return no data
+				$zero = 0;
+				$query->where($db->qn('s.update_site_id') . ' = :update_site_id');
```

Comment:
You don't need to bind a fix value just write it into the query.
```suggestion
				$query->where($db->quoteName('s.update_site_id') . ' = 0');
```
and remove the line before and after.